src/backend/drm/surface/gbm.rs

```diff
@@ -334,9 +340,10 @@ impl<A: AsRawFd + 'static> Drop for FbHandle<A> {
     }
 }
 
-fn attach_framebuffer<A>(drm: &Arc<DrmSurface<A>>, bo: &BufferObject<()>) -> Result<FbHandle<A>, Error>
+fn attach_framebuffer<A, D>(drm: &Arc<DrmSurface<D>>, bo: &BufferObject<()>) -> Result<FbHandle<D>, Error<A>>
 where
-    A: AsRawFd + 'static,
+    A: std::error::Error + Send + Sync,
```

Comment:
I don't think using `A` as a name for an error type is a good idea, especially when `A` represents the allocator in the rest of the file.

Also, this is pretty surprising that the caller can chose the error type when invoking this function, but I guess it's because the `Error::GbmError` is never constructed in it, right?

Reply:
Yes, also this function is private, so that is not really a concern for downstream.

But you are right about the generic name, I will change that.